File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.rb
# @param {String} s
# @return {Integer}
def first_uniq_char(s)
    hash = Hash.new(0)
    s.chars.each do |ch|
       hash[ch]+=1
    end

    s.chars.each_with_index do |ch, idx|
       return idx if hash[ch] == 1
    end
    -1
end

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    const obj = {};
    for(const x in s){
        if(s[x] in obj){
            obj[s[x]]+=1;
        }
        else{
            obj[s[x]]=1;
        }
    }
    console.log(obj)
    for(let i=0;i<s.length;i++){
        if(obj[s[i]]==1)return i;
    }
    return -1;
};